src/main.cpp

```diff
@@ -13,20 +13,37 @@ void resetQuickPowerCycleCount() {
   DPRINTF("Wrote QPC = %d\n", Settings->quick_power_cycle_count);
 }
 
+// this method both locks the board and will put it in recovery mode
 bool isInRecoveryMode() {
-  DPRINTF("Read QPC = %d\n", Settings->quick_power_cycle_count);
-  bool recovery = false;
-  if (Settings->quick_power_cycle_count > 1) {
-    recovery = true;
+  // no password set or board isn't armed, won't lock
+  if (strcmp(Settings->ap_self_password, "") == 0 ||
+      !Settings->board_lock_armed) {
```

Comment:
You should only check board_lock_armed if board_lock is on so the if statement should look like
"if(strlen(password) == 0 || !board_lock || !board_lock_armed)